packages/expo-router/src/link/preview/useScreenPreload.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  ParamListBase,
+  StackNavigationState,
+  type NavigationRoute,
+} from '@react-navigation/native';
+import { useCallback, useMemo, useState } from 'react';
+
+import { getParamsAndNodeFromHref } from './HrefPreview';
+import { store, type ReactNavigationState } from '../../global-state/router-store';
+import { useRouter } from '../../hooks';
+import { Href } from '../../types';
+import { resolveHref } from '../href';
+
+export function useScreenPreload(href: Href) {
+  const router = useRouter();
+  const [nextScreenId, setNextScreenId] = useState<string | undefined>();
+
+  const { params, routeNode } = useMemo(() => getParamsAndNodeFromHref(href), [href]);
+
+  const updateNextScreenId = useCallback((): void => {
+    const preloadedRoute = getPreloadedRouteFromRootStateByHref(href);
+
+    const routeKey = preloadedRoute?.key;
+
+    setNextScreenId(routeKey);
+  }, [params, routeNode]);
```

Comment:
Is href missing here? Also the API seems a bit odd. Passing href in to the function for computation instead of doing it externally and just having `[value, setValue]` as the return value